libclc/utils/prepare-builtins.cpp

```diff
@@ -104,6 +105,20 @@ int main(int argc, char **argv) {
     return 1;
   }
 
+  // AMDGPU remove incompatible functions pass replaces all uses of functions
+  // that use GPU features incompatible with the current GPU with null then
+  // deletes the function. This didn't use to cause problems, as all of libclc
+  // functions were inlined prior to incompatible functions pass. Now that the
+  // inliner runs later in the pipeline we have to remove all of the target
+  // features, so libclc functions will not be earmarked for deletion.
+  if (M->getTargetTriple().find("amdgcn") != std::string::npos) {
+    AttributeMask AM;
+    AM.addAttribute("target-features");
+    AM.addAttribute("target-cpu");
```

Comment:
Is there no way to just not generate these when creating the libclc IR?

Reply:
According to this comment: https://github.com/jchlanda/llvm/blob/jakub/pulldown_fix_libclc/libclc/CMakeLists.txt#L344 no. Once you specify the gfx version it will be embedded in the IR as `target-cpu`.

Reply:
Oh I see we need that for the clang built-ins to work